### PR TITLE
Fix go build process to actaully use customizes env variables

### PR DESCRIPTION
**Pull request by bryk**

### CODE REVIEW

**Comment by bryk** [comment]
@luxas Can you look at this?

**Comment by luxas** [comment]
Could you remove this also? 
```
// Scratch docker image is linux.
GOOS: 'linux',
```

It isn't required as `go` automatically will use `GOHOSTOS` and `GOHOSTARCH` (official builds will be on `linux/amd64`)
I will push some cross-compilation code soon

**Comment by bryk** [comment]
Makes sense, done. PTAL

File: build/backend.js [hunk 1]
```diff
@@ -66,8 +66,6 @@ gulp.task('backend:prod', ['package-backend-source'], function(doneFn) {
                 doneFn, {
                   // Disable cgo package. Required to run on scratch docker image.
                   CGO_ENABLED: '0',
```
Comment (luxas) [inline build/backend.js]: I hope it don't crashes with this extra `,` Does `js` allow that?

Comment (bryk) [inline build/backend.js]: It it okay. JS and golang allow (and like) trailing commas.

**Comment by luxas** [comment]
LGTM. The Travis build also passed

**Pull request merged**